File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
/*自定义服务*/
app.service("brandService",function ($http) {
    //分页查询
    this.findByPage=function(pageNum,pageSize) {
        return $http.get('../brand/findByPage.do?pageNum='+pageNum+'&pageSize='+pageSize);
    }
    //分页条按查询
    this.search=function(pageNum,pageSize,entity){
        return $http.post('../brand/search.do?pageNum='+pageNum+'&pageSize='+pageSize,entity);
    };

    //查询所有
    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    };
    this.add=function (entity) {
        return $http.post('../brand/add.do',entity)
    };
    this.update=function (entity) {
        return $http.post('../brand/update.do',entity)
    };
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    };
    this.dele=function (ids) {
        return $http.get('../brand/delete.do?ids='+ids);
    }

    this.findSelectList=function () {
        return $http.get('../brand/findSelectList.do');
    }
});

File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    List<TbBrand> findAll();
    PageResult findByPage(int pageNum, int pageSize);

    public void add(TbBrand tbBrand);

    /**
     * 根据id查询单个数据
     * @param id
     * @return
     */
    TbBrand findOne(Long id);

    /**
     * 更新品牌数据
     * @param tbBrand
     */
    void update(TbBrand tbBrand);

    /**
     * 删除选中数据
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 根据条件分页查询
     * @param pageNum
     * @param pageSize
     * @param tbBrand
     * @return
     */
    PageResult search(int pageNum, int pageSize, TbBrand tbBrand);

    /**
     * 查询所有商品信息，以下列选项要求的格式
     * @return
     */
    List<Map> findSelectList();


}
